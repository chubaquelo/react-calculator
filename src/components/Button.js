import React from 'react';
import PropTypes from 'prop-types';

export default function Button({ btnName, addClass }) {
  let btnStyle = 'border-white border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl';
  btnStyle += ` ${addClass}`;

  return <div className={btnStyle}>{btnName}</div>;
}

Button.propTypes = {
  btnName: PropTypes.string,
  addClass: PropTypes.string,
};

Button.defaultProps = {
  btnName: 'NONE',
  addClass: '',
};
