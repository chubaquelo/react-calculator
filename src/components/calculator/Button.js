import React from 'react';
import PropTypes from 'prop-types';

export default function Button({ btnName, addClass, updateCalc }) {
  let btnStyle = 'border-white select-none border-2 w-1/4 h-16 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl';
  btnStyle += ` ${addClass}`;

  return <button type="button" className={btnStyle} onClick={() => updateCalc(btnName)}>{btnName}</button>;
}

Button.propTypes = {
  btnName: PropTypes.string,
  addClass: PropTypes.string,
  updateCalc: PropTypes.func,
};

Button.defaultProps = {
  btnName: 'NONE',
  addClass: '',
  updateCalc: 0,
};
