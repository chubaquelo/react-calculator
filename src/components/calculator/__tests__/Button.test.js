import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../Button';

describe.only('Test Button Snapshot', () => {
  it('matches Button snapshot', () => {
    const forTest = renderer.create(<Button />).toJSON();
    expect(forTest).toMatchSnapshot();
  });
});
