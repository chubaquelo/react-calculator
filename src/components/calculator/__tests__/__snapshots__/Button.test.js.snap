// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button Snapshot matches Button snapshot 1`] = `
<button
  className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl "
  onClick={[Function]}
  type="button"
>
  NONE
</button>
`;
